v2/internal/crdmanagement/manager.go

```diff
@@ -399,6 +397,9 @@ func (m *Manager) applyCRDs(
 			return nil
 		})
 		if err != nil {
+			// Logging here before returning because we may cancel the lease (due to defer cancel()) above, and we want to ensure
```

Comment:
This change would be easy to miss. It's a simple logging fix because I had the operator crashing at this point and I had a hard time debugging it. This makes it better.